Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -197,6 +199,34 @@ template <class Type> class TableColumn : public API::Column {
   /// Re-arrange values in this column according to indices in indexVec
   void sortValues(const std::vector<size_t> &indexVec) override;
 
+  bool equals(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<TableColumn *>(otherColumn);
+    if (!convertedOtherColumn) {
+      return false;
+    }
+    auto type = this->type();
+    if (convertedOtherColumn->size() != this->size() ||
+        convertedOtherColumn->type() != type) {
+      return false;
+    }
+    auto otherData = convertedOtherColumn->data();
+    return compareVectors(otherData, tolerance);
+  }
+
+  bool equalsRelErr(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<TableColumn *>(otherColumn);
```

Comment:
Same comments above regarding the metadata/type checks. Can we move them to a private common function?